ditto/modify/system_structure.py

```diff
@@ -625,19 +628,17 @@ def set_nominal_voltages(self):
         # Now we take care of the Lines.
         # Since we should have the nominal voltage for every node (in a perfect world),
         # We just have to grab the nominal voltage of one of the end-points.
-        for obj in self.model.models:
-            # If we get a line
-            if isinstance(obj, Line) and obj.nominal_voltage is None:
-                # Get the from node
-                if hasattr(obj, "from_element") and obj.from_element is not None:
-                    node_from_object = self.model[obj.from_element]
+        for obj in self.model.iter_elements(Line, lambda x: x.nominal_voltage is None):
```

Comment:
@tarekelgindy Let me know what you think of the filter function. You may or may not find it more readable. It removes some of the heavily-nested blocks.